components/hab/src/command/pkg/bulkupload.rs

```diff
@@ -53,6 +54,8 @@ pub fn start(ui: &mut UI,
     let artifact_paths =
         vec_from_glob_with(&artifact_path.join("*.hart").display().to_string(), OPTIONS);
 
+    let pub_keys_paths = vec_from_glob_with(&key_path.join("*.pub").display().to_string(), OPTIONS);
```

Comment:
I know it's not directly involved in this change, but this line led me to the implementation of `vec_from_glob_with`, and I saw some things there that could merit a change.

In particular, we always pass the `OPTIONS` constant to that function, and these are the only two invocations. We may as well move `OPTIONS` to the implementation of `vec_from_glob_with` and simplify the implementation of `start`, as well as the invocation of `vec_from_glob_with`. It also keeps the complexity close to where it's needed.

Next up, the repetition of the `&<...>.display().to_string()` is kind of a bummer. Part of that is just the verbosity around getting to a `&str` (required by the `glob` crate) from a `PathBuf`. We could also shove this bit of ugliness down into `vec_from_glob_with`. One way would be to have `vec_from_glob_with` take an `AsRef<Path>` argument; then we could just pass in, e.g. `key_path.join("*.pub")`, and then call `&ARG.as_ref().display().to_string()`. You could also imagine an API that takes a directory and a str pattern, meaning "find all files matching this pattern in this directory". You could even go a step further and call it an extension.

That would hide the ugliness in `vec_from_glob_with`, and reduce the invocation to something like:
```rust
let pub_key_paths = vec_from_glob_with(key_path, "pub");
```
(At that point, I'd probably call it something like `paths_with_extension`.)

Then, once we get into the actual `vec_from_glob_with` implementation, we call `unwrap` _a lot_. Unless we can be absolutely certain that these `unwraps` are always safe, we should be a bit more circumspect.

Here's the current code:
```rust
    glob_with(pattern, options).unwrap()
                               .map(std::result::Result::unwrap)
                               .collect()
```
That could potentially panic on the initial `glob_with` call, as well as once for each item found (in the `map` call).

One nifty thing about the `Result` type is that it implements `FromIter`, meaning you can actually use `collect()` to get from a `Vec<Result<T,E>` to a `Result<Vec<T>, E>`! The first error found is returned and terminates the iteration.

(See https://doc.rust-lang.org/rust-by-example/error/iter_result.html for more.)

```rust
   glob_with(pattern, options)?.collect::<std::result::Result<Vec<_>, _>>()?
```
You'd need to return a Result, and possibly do some error mappings, but it would be safer overall.